lib/mvi/responses/base.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+module MVI
+  module Responses
+    class Base
+      attr_accessor :code, :query, :original_response
+
+      RESPONSE_CODES = {
+        success: 'AA',
+        failure: 'AE',
+        invalid_request: 'AR'
+      }.freeze
+
+      class << self
+        attr_accessor :endpoint
+      end
+
+      def self.mvi_endpoint(endpoint)
```

Comment:
Why does this exist?  It provides a way to set the Class variable `endpoint` which the code before it already does.
```ruby
class << self
  attr_accessor :endpoint
end
```